src/dotnet/MLSearch.Service/MLSearch.cs

```diff
@@ -1,9 +1,39 @@
 ﻿using ActualChat.Chat;
+using ActualChat.Users;
 
 namespace ActualChat.MLSearch;
 
-public class MLSearchImpl (ICommander commander): IMLSearch
+public class MLSearchImpl (IServiceProvider services): IMLSearch
```

Comment:
:( I tried to prevent service locator pattern in the ML module.
This makes code untestable.

Reply:
Yep. I understand, that explicit dependecies better than service locator. Fixed.